querydsl-core/src/main/java/com/querydsl/core/DefaultQueryMetadata.java

```diff
@@ -306,6 +301,9 @@ public void setLimit(Long limit) {
 
     @Override
     public void setModifiers(@Nullable QueryModifiers restriction) {
+        if (restriction == null) {
+            throw new NullPointerException();
```

Comment:
Then I think it's best to document this in `QueryBase.restrict()` and others, or annotate the argument as `@Nonnull` to help tools assist the user.